controllers/storagecluster/backingstorageclasses.go

```diff
@@ -0,0 +1,92 @@
+package storagecluster
+
+import (
+	"fmt"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	v1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type backingStorageClasses struct{}
+
+// ensureCreated ensures that backing storageclasses are created for the StorageCluster
+func (obj *backingStorageClasses) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	pvReclaimPolicy := corev1.PersistentVolumeReclaimDelete
+	allowVolumeExpansion := true
+	volumeBindingMode := v1.VolumeBindingWaitForFirstConsumer
+
+	for i := range sc.Spec.BackingStorageClasses {
+		bsc := &sc.Spec.BackingStorageClasses[i]
+		if bsc.Name == "" {
+			return reconcile.Result{}, fmt.Errorf("backingStorageClass name is empty")
+		}
+		storageClass := &storagev1.StorageClass{}
+		storageClass.ReclaimPolicy = &pvReclaimPolicy
+		storageClass.AllowVolumeExpansion = &allowVolumeExpansion
+		storageClass.VolumeBindingMode = &volumeBindingMode
+
+		storageClass.ObjectMeta = bsc.ObjectMeta
+		storageClass.Parameters = bsc.Parameters
+		storageClass.Provisioner = bsc.Provisioner
+		if bsc.Provisioner == "" {
+			platform, err := r.platform.getPlatform(r.Client)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			if platform != configv1.AWSPlatformType {
+				return reconcile.Result{}, fmt.Errorf("auto detection of provisioner is not supported for %s", platform)
+			}
+			storageClass.Provisioner = "ebs.csi.aws.com"
+		}
+
+		existingStorageClass := &storagev1.StorageClass{}
+		err := r.Client.Get(r.ctx, types.NamespacedName{Name: storageClass.Name}, existingStorageClass)
+		if errors.IsNotFound(err) {
+			// Since the StorageClass is not found, we will create a new one
+			err = r.Client.Create(r.ctx, storageClass)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		} else if err != nil {
+			return reconcile.Result{}, err
+		} else {
+			if !reflect.DeepEqual(storageClass.Parameters, existingStorageClass.Parameters) {
+				// Since we have to update the existing StorageClass
+				// So, we will delete the existing storageclass and create a new one
+				r.Log.Info("StorageClass needs to be updated, deleting it.", "StorageClass", storageClass.Name)
+				err = r.Client.Delete(r.ctx, existingStorageClass)
+				if err != nil {
+					r.Log.Error(err, "Failed to delete StorageClass.", "StorageClass", existingStorageClass.Name)
+					return reconcile.Result{}, err
+				}
+				r.Log.Info("Creating StorageClass.", "StorageClass", storageClass.Name)
+				err = r.Client.Create(r.ctx, storageClass)
+				if err != nil {
+					r.Log.Info("Failed to create StorageClass.", "StorageClass", storageClass.Name)
+					return reconcile.Result{}, err
+				}
+			}
+		}
```

Comment:
I can two potential issues with this code:
1. It allows the "update" of an existing storageclass even if it is in use!
2. It does not support the removal of a backing storage class from the list.

Reply:
@nb-ohad you mentioned you are going to discuss with @oritwas on this one, based on your discussion i will make required changes.

Reply:
@Madhu-1 Thay might be true for bullet #1, but you still have a gap for bullet #2